File: src/modifiers/highPriorityGesture.ts
import { createModifier } from './createModifier';
import type { GestureParams } from './gesture';

/// SwiftUI `.highPriorityGesture(_:)`. Attaches a gesture that takes
/// precedence over the view's own gestures, recognizing before them.
///
/// **Limited (v1):** same bridging constraints as `gesture` — only `'tap'`
/// and `'longPress'` are supported; arbitrary SwiftUI `Gesture` values are
/// not expressible across the bridge.
export function highPriorityGesture(params: GestureParams) {
  return createModifier('highPriorityGesture', params);
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/HighPriorityGesture.swift
import SwiftUI

/// SwiftUI `.highPriorityGesture(_:)`. Takes precedence over the view's own
/// gestures. Same v1 limitation as `gesture`: only `tap`/`longPress` bridge.
enum RNWHighPriorityGestureModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("highPriorityGesture") { view, params, ctx in
            RNWGestureSupport.apply(.highPriority, to: view, params: params, ctx: ctx)
        }
    }
}
